pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/ElementNode.java

```diff
@@ -80,6 +80,16 @@ public ElementNode(DocumentNode document,
         document.nodeToElementNode.put(node, this);
     }
 
+    private static int determineType(Node node) {
```

Comment:
```suggestion
    private static int determineType(Node node) {
        // As of PMD 6.48.0, only the experimental HTML module uses this naming 
        // convention to identify non-element nodes.
        // TODO PMD 7: maybe generalize this to other languages
```

In a general PMD 7 solution, I don't expect we use a naming convention to identify the node kind (we'd probably use an enum).